### PR TITLE
Add Inception Version 4

### PR DESCRIPTION
**Pull request by umesh-timalsina**
Adds a new Inception Version Model using prescriptions from the original redshift redshift paper as well as InceptionV4 Paper. All the `MaxPooling` layers have been replaced with Average Pooling
layers. At current form, some adjustments to the `Hyperparameters` are
needed before using it for `redshift` classification task.

@KyleAMoore Could we use some of the lessons from the `ResNet` and `InceptionV3` model to adjust the hyper-parameters for this InceptionV4?

### CODE REVIEW

**Comment by KyleAMoore** [comment]
Of the hyperparameters already included in the model, the only one that definitely will need to change is the input image shape, which should be `(64,64,5)` given no major changes to the preprocessing. The previous models did not include dropout, so I cannot comment on what would be a good value for the dropout chance. From my personal experimentation, I found little difference in the value of `num_redshift_bin`. Theoretically, any number greater than 1 should be usable, but it is likely that too low of values would have lower accuracy. Significant increases, such as the `1000` used here, do not seem to significantly affect accuracy in either direction as far as I have seen. I find that medium-sized powers of 2 (ex. `32` or `64`) to be both intuitive and sufficient, while likely being somewhat more time-efficient.

File: model/model.py [hunk 1]
```diff
@@ -232,3 +234,785 @@ def __str__(self):
...
+        The keep probability for the dropout layer
+    num_redshift_bins : int
+        The number of redshift classes for the end softmax layer
+
+    Notes
+    -----
+    This Network assumes that the data format is `channels_last`
+    """
+    def __init__(self,
+                 input_image_shape=(299, 299, 3),
```
Comment (KyleAMoore) [inline model/model.py:265]: I suggest changing the default value of `input_image_shape` to `(64,64,5)` to reflect the actual expected shape of photometric images.

Comment (umesh-timalsina) [reply model/model.py:265]: At this current form, An input image size of (64, 64, 5) is incompatible with the `stem-block` used here. This is because the stem block reduces the image size by almost 8 folds from (299, 299, 3) to (35 * 35 * 384). For a 64 * 64 * 5 image we would end up with ( 8 * 8 * 384). This is not logical to use as it would endup not being compatible with the `Reduction-A` and `Reduction-B` blocks down in the architecture.

Comment (KyleAMoore) [reply model/model.py:265]: That makes sense. My source of confusion right now is how would this be compatible with the images we are using? Will we need an additional dense layer between the input and the call to `add_stem` that reshapes the `(64,64,5)` image to the expected `(299,299,3)`?

Comment (umesh-timalsina) [reply model/model.py:265]: We should do some research on how to replace this stem block. The motivation behind using a these blocks is learning equivalent features with less parameters. In the origninal redshift paper, they have used a 5*5 kernel with an Average pooling block before using inception blocks. Not sure how to fair this one out.

Comment (umesh-timalsina) [reply model/model.py:265]: Basically, if you play with the parameters like `strides` and `num_filters` you can get an appropriate representation to feed forward to the inception blocks.

Comment (umesh-timalsina) [reply model/model.py:265]: https://www.kaggle.com/babbler/inception-model-for-fashion-mnist/notebook

This notebook has some interesting usage of inceptionV4 modules.
